File: Serene/Serene.Script/Northwind/Order/OrderDialog.cs

namespace Serene.Northwind
{
    using Common;
    using Serenity;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported]
    [IdProperty(OrderRow.IdProperty), NameProperty(OrderRow.Fields.OrderID), Flexify, Maximizable]
    [FormKey("Northwind.Order"), LocalTextPrefix("Northwind.Order"), Service("Northwind/Order")]
    public class OrderDialog : EntityDialog<OrderRow>
    {
        protected OrderForm form;

        public OrderDialog()
        {
            form = new OrderForm(this.IdPrefix);
        }

        protected override void LoadEntity(OrderRow entity)
        {
            base.LoadEntity(entity);
        }

        protected override List<ToolButton> GetToolbarButtons()
        {
            var buttons = base.GetToolbarButtons();

            buttons.Add(ReportHelper.CreateToolButton(new ReportButtonOptions
            {
                Title = "Invoice",
                CssClass = "export-pdf-button",
                ReportKey = "Northwind.OrderDetail",
                GetParams = () => new
                {
                    OrderID = this.EntityId
                }
            }));

            return buttons;
        }
    }
}

File: Serene/Serene.Web/Modules/Northwind/Order/OrderDialog.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class OrderDialog extends Serenity.EntityDialog<OrderRow, any> {
        protected getFormKey() { return OrderForm.formKey; }
        protected getIdProperty() { return OrderRow.idProperty; }
        protected getLocalTextPrefix() { return OrderRow.localTextPrefix; }
        protected getNameProperty() { return OrderRow.nameProperty; }
        protected getService() { return OrderService.baseUrl; }

        protected form = new OrderForm(this.idPrefix);

        constructor() {
            super();
        }

        getToolbarButtons() {
            var buttons = super.getToolbarButtons();

            buttons.push(Serene.Common.ReportHelper.createToolButton({
                title: 'Invoice',
                cssClass: 'export-pdf-button',
                reportKey: 'Northwind.OrderDetail',
                getParams: () => ({ OrderID: this.get_entityId() })
            }));

            return buttons;
        }

        protected updateInterface() {
            super.updateInterface();

            this.toolbar.findButton('export-pdf-button').toggle(this.isEditMode());
        }
    }
}